stubs/redis/redis/utils.pyi

```diff
@@ -1,8 +1,21 @@
-from typing import Any
+from typing import Any, ContextManager, Optional, Text, overload
+from typing_extensions import Literal
 
-HIREDIS_AVAILABLE: Any
+from .client import Pipeline, Redis
 
-def from_url(url, db=..., **kwargs): ...
-def pipeline(redis_obj): ...
+HIREDIS_AVAILABLE: Any
+@overload
+def from_url(url: Text, db: Optional[int] = ..., *, decode_responses: Literal[True], **kwargs) -> Redis[str]: ...
+@overload
+def from_url(url: Text, db: Optional[int] = ..., *, decode_responses: Literal[False] = ..., **kwargs) -> Redis[bytes]: ...
+@overload
+def str_if_bytes(value: bytes) -> str: ...
+@overload
+def str_if_bytes(value: object) -> object: ...
+@overload
+def safe_str(value: bytes) -> str: ...
+@overload
+def safe_str(value: object) -> str: ...
```

Comment:
The first overload is redundant, as `bytes` is a sub-type of `object` and the function returns the same type:

```suggestion
def safe_str(value: object) -> str: ...
```

Reply:
Great comment, I was too focused on the underlying use of `str_if_bytes` but the overload is not needed...